test/spec/controllers/partyCtrlSpec.js

```diff
@@ -33,6 +36,101 @@ describe("Party Controller", function() {
     });
   });
 
+  describe('initialization', function() {
+    beforeEach(function() {
+      sandbox.stub(rootScope, 'openModal');
+    });
+
+    context('party has 1 member', function() {
+      beforeEach(function() {
+        scope.group = { memberCount: 1 };
+      });
+
+      it('awards no new achievements', function() {
+        expect(User.set).to.not.be.called;
+        expect(rootScope.openModal).to.not.be.called;
+      });
+    });
+
+    context('party has 2 members', function() {
+      beforeEach(function() {
+        scope.group = { memberCount: 2 };
+      });
+
+      context('user does not have "Party Up" achievement', function() {
+        it('awards "Party Up" achievement', function() {
+          expect(User.set).to.be.calledOnce;
+          expect(User.set).to.be.calledWith(
+            { 'achievements.partyUp': true}
+          );
+          expect(rootScope.openModal).to.be.calledOnce;
+          expect(rootScope.openModal).to.be.calledWith(
+            'achievements/partyUp',
+            { controller: 'UserCtrl' }
```

Comment:
This will fail because it's being called with `{controller:'UserCtrl', size:'sm'}`. You could do:

```js
expect(rootScope.openModal).to.be.calledWith(
  'achievements/partyUp',
  sinon.match({ controller: 'UserCtrl' })
)
```

or just

```
expect(rootScope.openModal).to.be.calledWith('achievements/partyUp');
```

I don't think it's particularly important to know that the modal was opened with a particular controller, and, in fact, all the achievement modal stuff should probably be abstracted into a service. (not for this pr)